### ISSUE TITLE
When I start training the network, the training process gets stuck at the beginning and doesn't seem to start after a long time?

### ISSUE THREAD

**Issue by hj950815** [MIC-DKFZ/nnDetection#60]
### :question: Question

When I start training the network, the training process gets stuck at the beginning and doesn't seem to start after a long time?

my nndet_env :
----- PyTorch Information -----
PyTorch Version: 1.7.1+cu110
PyTorch Debug: False
PyTorch CUDA: 11.0
PyTorch Backend cudnn: 8005
PyTorch CUDA Arch List: ['sm_37', 'sm_50', 'sm_60', 'sm_70', 'sm_75', 'sm_80']
PyTorch Current Device Capability: (8, 6)
PyTorch CUDA available: True

----- System Information -----
System NVCC: nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2020 NVIDIA Corporation
Built on Tue_Sep_15_19:10:02_PDT_2020
Cuda compilation tools, release 11.1, V11.1.74
Build cuda_11.1.TC455_06.29069683_0

System Arch List: None
System OMP_NUM_THREADS: 2
System CUDA_HOME is None: True
System CPU Count: 64
Python Version: 3.8.12 | packaged by conda-forge | (default, Jan 30 2022, 23:53:36) 
[GCC 9.4.0]

----- nnDetection Information -----
det_num_threads 8
det_data is set True
det_models is set True

 Terminal multiple outputs during training part：
                                                                            
142 | model.head.regressor.conv_internal.c_internal0.norm  | GroupNorm               | 256    | [1, 128, 88, 20, 20]                                                                                                             | [1, 128, 88, 20, 20]                                                                                                            
143 | model.head.regressor.conv_internal.c_internal0.act   | ReLU                    | 0      | [1, 128, 88, 20, 20]                                                                                                             | [1, 128, 88, 20, 20]                                                                                                            
144 | model.head.regressor.conv_out                        | ConvGroupRelu           | 560 K  | [1, 128, 88, 20, 20]                                                                                                             | [1, 162, 88, 20, 20]                                                                                                            
145 | model.head.regressor.conv_out.conv                   | Conv3d                  | 560 K  | [1, 128, 88, 20, 20]                                                                                                             | [1, 162, 88, 20, 20]                                                                                                            
146 | model.head.regressor.scales                          | ModuleList              | 4      | ?                                                                                                                                | ?                                                                                                                               
147 | model.head.regressor.scales.0                        | Scale                   | 1      | [1, 162, 88, 20, 20]                                                                                                             | [1, 162, 88, 20, 20]                                                                                                            
148 | model.head.regressor.scales.1                        | Scale                   | 1      | [1, 162, 44, 10, 10]                                                                                                             | [1, 162, 44, 10, 10]                                                                                                            
149 | model.head.regressor.scales.2                        | Scale                   | 1      | [1, 162, 22, 5, 5]                                                                                                               | [1, 162, 22, 5, 5]                                                                                                              
150 | model.head.regressor.scales.3                        | Scale                   | 1      | [1, 162, 11, 5, 5]                                                                                                               | [1, 162, 11, 5, 5]                                                                                                              
151 | model.head.regressor.loss                            | GIoULoss                | 0      | ?                                                                                                                                | ?                                                                                                                               
152 | model.anchor_generator                               | AnchorGenerator3DS      | 0      | [[1, 1, 352, 80, 80], [[1, 128, 88, 20, 20], [1, 128, 44, 10, 10], [1, 128, 22, 5, 5], [1, 128, 11, 5, 5]]]                      | [[1091475, 6]]                                                                                                                  
153 | model.segmenter                                      | DiCESegmenterFgBg       | 66     | [[1, 32, 352, 80, 80], [1, 64, 176, 40, 40], [1, 128, 88, 20, 20], [1, 128, 44, 10, 10], [1, 128, 22, 5, 5], [1, 128, 11, 5, 5]] | ?                                                                                                                               
154 | model.segmenter.conv_out                             | ConvInstanceRelu        | 66     | [1, 32, 352, 80, 80]                                                                                                             | [1, 2, 352, 80, 80]                                                                                                             
155 | model.segmenter.conv_out.conv                        | Conv3d                  | 66     | [1, 32, 352, 80, 80]                                                                                                             | [1, 2, 352, 80, 80]                                                                                                             
156 | model.segmenter.dice_loss                            | SoftDiceLoss            | 0      | ?                                                                                                                                | ?                                                                                                                               
157 | model.segmenter.dice_loss.nonlin                     | Softmax                 | 0      | ?                                                                                                                                | ?                                                                                                                               
158 | model.segmenter.ce_loss                              | CrossEntropyLoss        | 0      | ?                                                                                                                                | ?                                                                                                                               
159 | model.segmenter.logits_convert_fn                    | Softmax                 | 0      | ?                                                                                                                                | ?                                                                                                                               
160 | pre_trafo                                            | Compose                 | 0      | ?                                                                                                                                | ?                                                                                                                               
161 | pre_trafo.transforms                                 | ModuleList              | 0      | ?                                                                                                                                | ?                                                                                                                               
162 | pre_trafo.transforms.0                               | FindInstances           | 0      | ?                                                                                                                                | ?                                                                                                                               
163 | pre_trafo.transforms.1                               | Instances2Boxes         | 0      | ?                                                                                                                                | ?                                                                                                                               
164 | pre_trafo.transforms.2                               | Instances2Segmentation  | 0      | ?                                                                                                                                | ?                                                                                                                               
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
18.9 M    Trainable params
0         Non-trainable params
18.9 M    Total params
75.477    Total estimated model params size (MB)
Validation sanity check: 0it [00:00, ?it/s]INFO Using validation DataLoader3DOffset with {}
INFO Building Sampling Cache for Dataloder
Sampling Cache: 100%|█████████████████████████████████████████████████████████████████████████████████| 84/84 [00:00<00:00, 34994.19it/s]
INFO Using 7 num_processes and 2 num_cached_per_queue for augmentation.                                           | 0/84 [00:00<?, ?it/s]
INFO VALIDATION KEYS:
 odict_keys(['breast_011', 'breast_012', 'breast_019', 'breast_020', 'breast_021', 'breast_023', 'breast_028', 'breast_029', 'breast_032', 'breast_036', 'breast_039', 'breast_041', 'breast_042', 'breast_047', 'breast_049', 'breast_051', 'breast_053', 'breast_056', 'breast_069', 'breast_074', 'breast_085', 'breast_091', 'breast_098', 'breast_100', 'breast_101', 'breast_111', 'breast_116', 'breast_129', 'breast_131', 'breast_133', 'breast_135', 'breast_136', 'breast_138', 'breast_149', 'breast_155', 'breast_161', 'breast_165', 'breast_169', 'breast_170', 'breast_185', 'breast_191', 'breast_196', 'breast_197', 'breast_198', 'breast_216', 'breast_218', 'breast_221', 'breast_224', 'breast_231', 'breast_235', 'breast_238', 'breast_242', 'breast_245', 'breast_256', 'breast_259', 'breast_266', 'breast_267', 'breast_275', 'breast_276', 'breast_277', 'breast_282', 'breast_287', 'breast_292', 'breast_301', 'breast_306', 'breast_315', 'breast_318', 'breast_324', 'breast_328', 'breast_336', 'breast_341', 'breast_349', 'breast_360', 'breast_373', 'breast_386', 'breast_387', 'breast_389', 'breast_391', 'breast_399', 'breast_402', 'breast_403', 'breast_408', 'breast_418', 'breast_420'])
Validation sanity check:   0%|                                                                                    | 0/10 [00:00<?, ?it/s]using pin_memory on device 0

Can you give me some advice,?   I ran 'nndet_prep --full_check' and it didn't seem to be a problem.

**Comment by mibaumgartner** [comment]
Dear @hj950815 ,

setting `OMP_NUM_THREADS = 1` is always required otherwise training gets stuck. Since this Issue came up in the past as well, I just added a new section to the readme. Please check out the `FAQ and Common Issues` section. If the problem persists and the readme didn't help, please let me know.

Best,
Michael

**Comment by hj950815** [comment]
Thank you very much for your help, I took it for granted that OMP_NUM_THREADS = 2, which caused this problem. 
Thanks again for your answer！
